### PR TITLE
Add Font Awesome support for Pages links

### LINKED ISSUE #122: [feature/3.2] Add custom FontAwesome pages icon
Add the ability to define custom FontAwesome icon in ACP for every custom page (boardrules-like) for phpBB 3.2 extension version.

### PR DESCRIPTION
**Pull request by VSEphpbb**
Fixes #122

### CODE REVIEW

File: entity/page.php [hunk 1]
```diff
@@ -482,6 +484,48 @@ public function set_template($template)
 	}
 
 	/**
+	* Get page icon font name
```
Comment (senky) [inline entity/page.php]: Why don't you just `{@inheritdoc}`?

File: event/listener.php [hunk 2]
```diff
@@ -118,23 +115,13 @@ public function show_page_links()
...
-			}
-
 			// Assign template var data
 			$this->template->assign_block_vars($row['page_link_event_name'] . '_links', array(
 				'U_LINK_URL' => $this->helper->route('phpbb_pages_dynamic_route_' . $row['page_id']),
 				'LINK_ROUTE' => $row['page_route'],
 				'LINK_TITLE' => $row['page_title'],
-				'ICON_LINK' => $custom_icon,
+				'ICON_FONT'  => $row['page_icon_font'],
+				'ICON_LINK'  => !$row['page_icon_font'] ? $this->page_operator->get_page_icon($row['page_route']) : '',
```
Comment (senky) [inline event/listener.php]: I also like shorthand for `!= ''`, but I remember from the old times that in my MODs I was instructed to use `!empty()` instead. Maybe this is the case as well?

File: tests/entity/page_entity_import_test.php [hunk 3]
```diff
@@ -114,6 +115,16 @@ public function import_test_fail_data()
 			'page_description'	=> str_repeat('a', 256),
 		));
 
+		// Too long
+		$data[] = array_merge($import_data[1], array(
```
Comment (senky) [inline tests/entity/page_entity_import_test.php]: This looks like not belonging here.

File: entity/page.php (hunk 1 above)
Comment (VSEphpbb) [inline entity/page.php]: That would be a job for a separate PR, as it would need to be done extension-wide.

File: tests/entity/page_entity_import_test.php (hunk 3 above)
Comment (VSEphpbb) [inline tests/entity/page_entity_import_test.php]: it's corrected now

File: adm/style/manage_pages.html [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<!-- INCLUDECSS ./../assets/css/font-awesome.min.css -->
```
Comment (rxu) [inline adm/style/manage_pages.html]: Also, it looks like it's already included in ```simple_header.html```.

**Comment by VSEphpbb** [comment]
It already does live hinting

**Comment by rxu** [comment]
Really? Didn't notice one.

**Comment by VSEphpbb** [comment]
It's there... See the JS

**Comment by rxu** [comment]
Oh. Then it doesn't work for me. No JS errors in the console though.

**Comment by VSEphpbb** [comment]
@rxu oh, we misunderstood each other. what you are asking about is available when opening the Font Aweseom window that is linked to in the field's label.

File: event/listener.php (hunk 2 above)
Comment (VSEphpbb) [inline event/listener.php]: the conditional test used here is fine. all the others would give the exact same result with the data possible here, and this is the most concise of them.
